pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -269,36 +270,37 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 	masterIgnSecret := IgnitionSecret(masterIgnAsset.Files()[0].Data, clusterID.InfraID, "master")
 	machineManifests = append(machineManifests, bootstrapIgnSecret, masterIgnSecret)
 
-	timer.StartTimer(machineStage)
 	// Create the machine manifests.
+	timer.StartTimer(machineStage)
+	machineNames := []string{}
+
 	for _, m := range machineManifests {
 		m.SetNamespace(capiutils.Namespace)
 		if err := cl.Create(ctx, m); err != nil {
 			return fileList, fmt.Errorf("failed to create control-plane manifest: %w", err)
 		}
 		i.appliedManifests = append(i.appliedManifests, m)
+
+		if machine, ok := m.(*clusterv1.Machine); ok {
+			machineNames = append(machineNames, machine.Name)
```

Comment:
What about:
```suggestion
		if m.GetObjectKind().GroupVersionKind().Kind == "Machine" {
			machineNames = append(machineNames, m.GetName())
```
Not sure which way is considered more "idiomatic".

Reply:
Me neither. My first instinct was to go with something like your suggestion, but wasn't certain of how to do it. When testing this suggestion, none of the machines were added to the checks.